### PR TITLE
TMA-537: Release Brick failing with production_tag

### PR DESCRIPTION
**Pull request by vhtien**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Changelog
- [ ] Build passing

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/8/ : SUCCESS in 4s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/842/ : SUCCESS in 33s

**Comment by korczis** [comment]
exteded test

**Comment by korczis** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/11/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/795/ : FAILURE in 46m 33s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/567/ : UNSTABLE in 14m 01s

**Comment by korczis** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/12/ : SUCCESS in 4s
- https://ci.intgdc.com/job/gooddata-ruby-tests-zuul/796/ : SUCCESS in 45m 40s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/568/ : UNSTABLE in 15m 38s

**Comment by vhtien** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/34/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/861/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/8/ : SUCCESS in 2m 53s

**Comment by vhtien** [comment]
extended test

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/37/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/810/ : SUCCESS in 41m 34s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/583/ : UNSTABLE in 14m 43s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/46/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/884/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/31/ : SUCCESS in 2m 57s

File: lib/gooddata/lcm/actions/collect_meta.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ class CollectMeta < BaseAction
 
       PARAMS = define_params(self) do
         description 'Production Tag Name'
-        param :production_tag, instance_of(Type::StringType), required: false
+        param :production_tags, instance_of(Type::StringType), required: false
```
Comment (panjan) [inline lib/gooddata/lcm/actions/collect_meta.rb]: We should maintain backward compatibility (i.e. support both `production_tag` and `production_tags`).

File: lib/gooddata/lcm/actions/collect_tagged_objects.rb [hunk 2]
```diff
@@ -34,15 +34,15 @@ def call(params)
           results = []
           segments_to_tags = Helpers.segment_production_tags(params.segments)
           transfer_all = GoodData::Helpers.to_boolean(params.transfer_all)
-          return results unless params.production_tag || segments_to_tags.any? || transfer_all
+          return results unless params.production_tags || segments_to_tags.any? || transfer_all
```
Comment (panjan) [inline lib/gooddata/lcm/actions/collect_tagged_objects.rb]: We should also check `params.production_tags` for emptiness.

File: lib/gooddata/lcm/helpers/tags_helper.rb [hunk 3]
```diff
@@ -27,7 +27,8 @@ def parse_production_tags(production_tags, segment_production_tags)
           separator = ','
           tags = segment_production_tags || production_tags
           return [] unless tags
-          tags.split(separator).map(&:strip)
+          tags = tags.split(separator).map(&:strip) unless tags.is_a?(Array)
```
Comment (panjan) [inline lib/gooddata/lcm/helpers/tags_helper.rb]: Please add a unit test for the case when there are multiple tags.

File: lib/gooddata/lcm/actions/collect_meta.rb (hunk 1 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/collect_meta.rb]: I mentioned it in the ticket. Waiting for other feedbacks.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/53/ : SUCCESS in 5s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/891/ : SUCCESS in 14s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/38/ : SUCCESS in 12s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/54/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/892/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/39/ : SUCCESS in 12s

File: lib/gooddata/lcm/helpers/tags_helper.rb (hunk 3 above)
Comment (vhtien) [inline lib/gooddata/lcm/helpers/tags_helper.rb]: done

File: lib/gooddata/lcm/actions/collect_tagged_objects.rb (hunk 2 above)
Comment (vhtien) [inline lib/gooddata/lcm/actions/collect_tagged_objects.rb]: done

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/55/ : SUCCESS in 4s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/893/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/40/ : SUCCESS in 11s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/85/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/929/ : SUCCESS in 54s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/75/ : FAILURE in 18s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/88/ : SUCCESS in 6s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/933/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/79/ : SUCCESS in 24s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/38/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/936/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/82/ : SUCCESS in 11s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/848/ : SUCCESS in 45m 39s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/621/ : UNSTABLE in 19m 43s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/39/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/937/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/83/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/850/ : SUCCESS in 45m 22s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/623/ : FAILURE in 3m 07s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/40/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-rubocop-zuul/938/ : SUCCESS in 21s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/84/ : SUCCESS in 15s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/851/ : SUCCESS in 45m 22s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/624/ : SUCCESS in 23m 19s

**Pull request merged**